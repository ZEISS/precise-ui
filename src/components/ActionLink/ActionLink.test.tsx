import * as React from 'react';
import * as enzyme from 'enzyme';
import { ActionLink } from './';

describe('<ActionLink />', () => {
  it('should render an empty <ActionLink> component', () => {
    const wrapper = enzyme.shallow(<ActionLink />);
    expect(wrapper.text()).toBe('<Styled(Anchor) />');
  });

  it('should render a <ActionLink> component with some content', () => {
    const wrapper = enzyme.mount(<ActionLink>Content</ActionLink>).mount();
    expect(wrapper.text()).toBe('Content');
  });

  it('should render an active <ActionLink> with some content', () => {
    const wrapper = enzyme.shallow(<ActionLink active>Content</ActionLink>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render an active disabled <ActionLink> with some content', () => {
    const wrapper = enzyme.shallow(
      <ActionLink disabled active>
        Content
      </ActionLink>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a <ActionLink> with some content component', () => {
    const wrapper = enzyme.shallow(
      <ActionLink href="foo">
        <div>Content</div>
      </ActionLink>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should execute the click handler if enabled', () => {
    let clicked = false;
    const clickHandler = () => {
      clicked = true;
    };
    const wrapper = enzyme.mount(<ActionLink onClick={clickHandler}>Content</ActionLink>);
    wrapper.simulate('click', {
      preventDefault() {},
    });
    expect(clicked).toBe(true);
  });

  it('should not execute the click handler if disabled', () => {
    let clicked = false;
    const clickHandler = () => {
      clicked = true;
    };
    const wrapper = enzyme.mount(
      <ActionLink onClick={clickHandler} disabled>
        Content
      </ActionLink>,
    );
    wrapper.simulate('click', {
      preventDefault() {},
    });
    expect(clicked).toBe(false);
  });

  it('should not fail if no history given', () => {
    let clicked = false;
    const clickHandler = () => {
      clicked = true;
    };
    const wrapper = enzyme.mount(
      <ActionLink to="/foo" onClick={clickHandler}>
        Content
      </ActionLink>,
    );
    wrapper.simulate('click', {
      preventDefault() {},
    });
    expect(clicked).toBe(true);
  });
});
