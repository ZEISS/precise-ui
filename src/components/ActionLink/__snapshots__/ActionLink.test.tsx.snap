// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionLink /> should render a <ActionLink> with some content component 1`] = `
<Styled(Anchor)
  href="foo"
>
  <div>
    Content
  </div>
</Styled(Anchor)>
`;

exports[`<ActionLink /> should render an active <ActionLink> with some content 1`] = `
<Styled(Anchor)
  active={true}
>
  Content
</Styled(Anchor)>
`;

exports[`<ActionLink /> should render an active disabled <ActionLink> with some content 1`] = `
<Styled(Anchor)
  active={true}
  disabled={true}
>
  Content
</Styled(Anchor)>
`;
